### PR TITLE
Raise helpful error if dict passed to scoring callback

### PR DESCRIPTION
**Pull request by lofifnc**
Introduces fix for issue #542 informing the user of the incompatibility in regards to multi-metric evaluation with scikit-learn.

### CODE REVIEW

File: skorch/callbacks/scoring.py [hunk 1]
```diff
@@ -81,6 +81,7 @@ def __init__(
     def _get_name(self):
         """Find name of scoring function."""
         if self.name is not None:
+            print("name", self.name)
```
Comment (BenjaminBossan) [inline skorch/callbacks/scoring.py]: Could you remove this please? This is probably also the reason the tests fail.

**Comment by BenjaminBossan** [comment]
Thanks for the contribution, Alex. Removing the debugging print statement should hopefully fix the tests.

File: skorch/callbacks/scoring.py (hunk 1 above)
Comment (lofifnc) [reply skorch/callbacks/scoring.py]: yea my bad

**Comment by BenjaminBossan** [comment]
Could you please also check the error message within the unit test? `ValueError` is very general and could also be triggered by something else.

File: skorch/tests/callbacks/test_scoring.py [hunk 2]
```diff
@@ -457,6 +457,22 @@ def test_multiple_scorings_share_cache(
         for c1, c2 in zip(cbs['a1'].y_trues_, cbs['a2'].y_trues_):
             assert id(c1) == id(c2)
 
+    def test_multiple_scorings_with_dict(
+            self, net_cls, module_cls, train_split, caching_scoring_cls, data):
```
Comment (thomasjpfan) [inline skorch/tests/callbacks/test_scoring.py]:
```suggestion
            self, net_cls, module_cls, train_split, scoring_cls, data):
```

File: skorch/tests/callbacks/test_scoring.py [hunk 3]
```diff
@@ -457,6 +457,22 @@ def test_multiple_scorings_share_cache(
...
+        # This test checks if an exception is raised when a dictionary is passed as scorer.
+        net = net_cls(
+            module=module_cls,
+            callbacks=[
+                caching_scoring_cls({'a1': 'accuracy', 'a2': 'accuracy'}),
```
Comment (thomasjpfan) [inline skorch/tests/callbacks/test_scoring.py]:
```suggestion
                scoring_cls({'a1': 'accuracy', 'a2': 'accuracy'}),
```

**Comment by BenjaminBossan** [comment]
Thank you very much.

I didn't know about the `match` keyword in `pytest.raises`, we could actually update quite a few of our tests this way.

**Pull request merged**